pkg/cmd/pipelinerun/logs.go

```diff
@@ -0,0 +1,269 @@
+// Copyright © 2019 The Knative Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipelinerun
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/logs"
+	v1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/reconciler/v1alpha1/taskrun/resources"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	typedv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const (
+	msgPipelineNotFoundErr = "Error in retrieving Pipeline"
+	msgPRNotFoundErr       = "Error in retrieving PipelineRun"
+)
+
+type taskRunMap map[string]*v1alpha1.PipelineRunTaskRunStatus
+
+type taskRun struct {
+	name   string
+	status v1alpha1.TaskRunStatus
+}
+
+type PipelineRunLogs struct {
```

Comment:
I think I would split this struct into several pieces : the options (initsteps, follow, …), the i/o (out, err) and the client (on it's own)